### DISCUSSION TITLE
useComboBox and --trigger-width

### DISCUSSION THREAD

**Discussion by luansantosti** [adobe/react-spectrum#5656] [Q&A] [answered]
Hey guys, I'm building a combobox using the useComboBox hook.
by following the docs, I noticed one thing: the width of my popover/listbox is not the same as the trigger, checking the docs for the combobox component, I saw that there is a css variable `--trigger-width` to handle this case, but what about for the useCombobox hook? do we have something similar?

btw, I could use a fixed width, but this is not the case since it could change based on the viewport.

Here is a screenshot:
![Screen Shot 2024-01-04 at 11 23 41](https://github.com/adobe/react-spectrum/assets/6179731/c677a557-f457-466a-a151-ae620cc7ea64)

thanks a lot in advance!

**Comment by reidbarber** [comment, marked as answer]
There isn't one included, but you could include the same feature in your component:

https://github.com/adobe/react-spectrum/blob/f9143723fe44a1c2e03eaec93f1cc2dba4e020cf/packages/react-aria-components/src/ComboBox.tsx#L148-L159

https://github.com/adobe/react-spectrum/blob/f9143723fe44a1c2e03eaec93f1cc2dba4e020cf/packages/react-aria-components/src/ComboBox.tsx#L197

**Reply by luansantosti** [reply]
Got it, I'll do that, thanks @reidbarber!
